src/Level Bars/main.vala

```diff
@@ -0,0 +1,60 @@
+#! /usr/bin/env -S vala workbench.vala --pkg gtk4 --pkg libadwaita-1
+
+Gtk.LevelBar bar_discrete;
+
+public void main () {
+
+    var bar_continuous = (Gtk.LevelBar) workbench.builder.get_object ("bar_continuous");
+
+    bar_continuous.add_offset_value ("full", 100);
+    bar_continuous.add_offset_value ("half", 50);
+    bar_continuous.add_offset_value ("low", 25);
+
+    var bar_discrete = (Gtk.LevelBar) workbench.builder.get_object ("bar_discrete");
+
+    bar_discrete.add_offset_value ("very-weak", 1);
+    bar_discrete.add_offset_value ("weak", 2);
+    bar_discrete.add_offset_value ("moderate", 4);
+    bar_discrete.add_offset_value ("strong", 6);
+
+
+    var entry = (Gtk.Entry) workbench.builder.get_object ("entry");
+
+    var label_strength = (Gtk.Label) workbench.builder.get_object ("label_strength");
+
+    entry.notify["text"].connect(estimatePasswordStrength);
+}
+
+// This is not a secure way to estimate password strength
+// Use appropriate solutions instead
+// such as https://github.com/dropbox/zxcvbn
+
+public void estimatePasswordStrength () {
```

Comment:
```suggestion
public void estimate_password_strength () {
```

Reply:
I have made the following implementations, however I am running into an issue while compiling where entry and bar_discrete do not exist inside estimate_password_strength() despite the connection from entry.notify["text"].connect(estimate_password_strength).